contribs/drt/src/main/java/org/matsim/contrib/drt/analysis/DrtRequestAnalyzer.java

```diff
@@ -179,4 +196,24 @@ public void handleEvent(PassengerDroppedOffEvent event) {
 			performedRequestSequences.get(event.getRequestId()).droppedOff = event;
 		}
 	}
+
+	@Override
+	public void handleEvent(PersonMoneyEvent event) {
+		if (event.getTransactionPartner() != null && event.getTransactionPartner().equals(mode) &&
```

Comment:
After a series of refactorings, `DrtRequestAnalyzer` has changed its purpose - it's mostly meant for collecting and grouping request-related events, while the analysis part is done elsewhere (sorry for keeping the old class name - it should be renamed).

It may be reasonable to attache `PersonMoneyEvent drtFare` events to event sequences, but `double sumFaresNotReferencingATrip` at the top level feels a bit unconnected to this class.

Reply:
Maybe collecting non-request-related money events into a list for later postprocessing would be a good compromise?

Reply:
I understand that argument and I had that same impression that `sumFaresNotReferencingATrip` is a little weird here, but I liked the idea of listening to PersonMoneyEvents only once and to attach the fare to the drt request `DrtRequestAnalyzer` seemed an obvious place. I can have `DrtRequestAnalyzer` filter and collect all 'PersonMoneyEvents' referencing a drt fare of that drt operator and sum those fares in `DrtAnalysisControlerListener` which writes the `drt_customer_stats.csv`. I will work on this now.

Reply:
done. @michalmac please have a look. I found that DrtLegsAnalyzer is a good fit because it prepares almost all data for the drt_customer_stats, but maybe DrtAnalysisControlerListener is better because those PersonMoneyEvents are no longer related to legs. Still DrtAnalysisControlerListener is a bit clunky with lots of small bits of code to put together data for different output statistics. I felt maybe it's better not to add more actual calculations to it.

Reply:
That's awesome! Thanks!